tests/msc4140/delayed_event_test.go

```diff
@@ -466,6 +467,15 @@ func TestDelayedEvents(t *testing.T) {
 		time.Sleep(1 * time.Second)
 		deployment.StartServer(t, hsName)
 
+		// The rest of the test assumes the second delayed event (10 second delay) still has
+		// hasn't been sent yet.
+		if time.Now().Sub(beforeScheduleStateTimestamp2) > 10*time.Second {
+			t.Fatalf(
+				"Test took too long to run, cannot guarantee delayed event timings. " +
+					"More than 10 seconds elapsed between scheduling the delayed event and now when we're about to check for it.",
+			)
+		}
```

Comment:
Actually fixing this and making this kind of test consistent is tricky.

If we just increase the delay time, then the test will just take longer to run even in the fast scenarios. Ideally, the second delayed event should be delayed by at-least `COMPLEMENT_SPAWN_HS_TIMEOUT_SECS` (defaults to 30 seconds) since we're stopping and starting servers here 🤔 - although we use [`COMPLEMENT_SPAWN_HS_TIMEOUT_SECS=120`](https://github.com/element-hq/synapse/blob/f4320b5a4926a8f5b7fb7d6899a22f7dc7236114/scripts-dev/complement.sh#L277-L280) with the Synapse workers variant and two minutes sounds painful to wait for.